src/map/skill.c

```diff
@@ -13385,10 +13385,10 @@ static int skill_unit_onplace_timer(struct skill_unit *src, struct block_list *b
 		case UNT_MANHOLE:
 			if (sg->val2 == 0 && tsc && (sg->unit_id == UNT_ANKLESNARE || bl->id != sg->src_id)) {
 				int sec = skill->get_time2(sg->skill_id, sg->skill_lv);
-				if (bl->type == BL_MOB && map->list[bl->m].flag.gvg_castle == 0) {
+				if (sg->unit_id == UNT_MANHOLE && bl->type == BL_MOB) {
 					struct mob_data *md = BL_UCAST(BL_MOB, bl);
-					if (sg->unit_id == UNT_MANHOLE && (md->class_ == MOBID_EMPELIUM || md->class_ == MOBID_BARRICADE || md->class_ == MOBID_S_EMPEL_1 || md->class_ == MOBID_S_EMPEL_2)) {
-						// Do nothing...
+					if (md->class_ == MOBID_EMPELIUM || md->class_ == MOBID_BARRICADE || md->class_ == MOBID_S_EMPEL_1 || md->class_ == MOBID_S_EMPEL_2) {
+						// Do nothing if target are the specified monsters on manhole.
```

Comment:
what is the purpose of this inner block ?  it doesn't seem to cause any side effects so there seems to be no pint in the inner if